pandas/core/groupby/ops.py

```diff
@@ -914,18 +916,22 @@ def agg_series(
         np.ndarray or ExtensionArray
         """
 
-        if not isinstance(obj._values, np.ndarray):
+        if not isinstance(obj._values, np.ndarray) and not isinstance(
+            obj._values, ArrowExtensionArray
+        ):
             # we can preserve a little bit more aggressively with EA dtype
             #  because maybe_cast_pointwise_result will do a try/except
             #  with _from_sequence.  NB we are assuming here that _from_sequence
             #  is sufficiently strict that it casts appropriately.
             preserve_dtype = True
 
         result = self._aggregate_series_pure_python(obj, func)
-
         npvalues = lib.maybe_convert_objects(result, try_float=False)
+
         if preserve_dtype:
             out = maybe_cast_pointwise_result(npvalues, obj.dtype, numeric_only=True)
+        elif isinstance(obj._values, ArrowExtensionArray):
+            out = maybe_cast_to_pyarrow_dtype(result, npvalues)
```

Comment:
If the cast is successful, I think we never use `npvalues`, is that right? In that case, I think we want to avoid the computation that goes into them.

Reply:
resolved, thanks